pkg/sshd/server.go

```diff
@@ -258,7 +258,7 @@ func (s *server) answer(channel ssh.Channel, requests <-chan *ssh.Request, sshCo
 					// TODO: this is an important error case that needs to be covered
 					// Probably the best solution is to change the lock into a lease so that even on unlock failures, RepositoryLock will eventually yield
 				}
-				s.cleanerRef.Unlock()
+				s.cleanerRef.RUnlock()
```

Comment:
I believe you also need to change the [`Unlock` call above](https://github.com/deis/builder/pull/209/files#diff-b60cdeda68b99d71a51ec60fc7298c49R244) to `RUnlock`